### PR TITLE
Optimize CI tests

### PR DESCRIPTION
**Pull request by Feiyang1**
- Removed integration test projects that are not really useful
- Split `Test Modified Packages` workflow into the following workflows
  - Test Modified Packages (Test all modified packages except the following ones)
  - Firestore
  - Firestore Integration
  - FCM Integration
  - Auth
  - rxFire, @firebase/testing, @firebase/rules-unit-testing
  - firebase-namespace-integration-test
- Only build the packages needed for running tests

### CODE REVIEW

File: .github/workflows/test-changed-firestore.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      run: |
+        sudo apt-get update
+        sudo apt-get install google-chrome-stable
+    - name: Test setup and yarn install
+      run: |
+        cp config/ci.config.json config/project.json
+        yarn
+    - name: yarn build
+      run: yarn build
+    - name: Run tests on changed packages
```
Comment (hsubox76) [inline .github/workflows/test-changed-firestore.yml:30]: "Run tests if firestore package has changed"?

File: scripts/run_changed.js [hunk 2]
```diff
@@ -24,23 +24,21 @@ const root = resolve(__dirname, '..');
 const git = simpleGit(root);
 
 // use test:ci command in CI
-const testCommand = !!process.env.CI ? 'test:ci' : 'test';
+// const testCommand = !!process.env.CI ? 'test:ci' : 'test';
+const testCommand = 'test:ci';
```
Comment (hsubox76) [inline scripts/run_changed.js:28]: revert?

File: scripts/run_changed.js [hunk 3]
```diff
@@ -49,31 +47,36 @@ const fullTestTriggerFiles = [
...
 /**
  * These files trigger tests in other dirs
  */
 const specialPaths = {
-  'scripts/emulator-testing/emulators/firestore-emulator.ts': [
-    '@firebase/firestore'
-  ],
+  // 'scripts/emulator-testing/emulators/firestore-emulator.ts': [
+  //   '@firebase/firestore'
+  // ],
```
Comment (hsubox76) [inline scripts/run_changed.js:59]: delete when done?

File: scripts/run_changed.js [hunk 4]
```diff
@@ -88,6 +91,7 @@ async function getChangedPackages() {
   const changedFiles = diff.split('\n');
   const changedPackages = {};
   for (const filename of changedFiles) {
+    console.log(filename);
```
Comment (hsubox76) [inline scripts/run_changed.js:94]: delete?

File: scripts/run_changed_firestore.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { resolve } from 'path';
+import { spawn, exec } from 'child-process-promise';
+import chalk from 'chalk';
+import simpleGit from 'simple-git/promise';
+
+interface TestTask {
```
Comment (hsubox76) [inline scripts/run_changed_firestore.ts:23]: Is this typescript makeover supposed to be applied to run_changed too?

**Review by schmidt-sebastian — COMMENTED** [review]
Not my area of expertise, so I will defer to Christina. Idea LGTM.

File: scripts/ci-test/run_changed.ts [hunk 6]
```diff
@@ -133,60 +156,43 @@ async function getChangedPackages() {
...
-  if (Object.keys(changedPackages).length > 0) {
-    return {
-      testAll: false,
-      changedPackages
-    };
-  } else {
+  if (testTasks.length === 0) {
     console.log(
-      chalk`{green No changes detected in any package. Skipping all package-specific tests.}`
+      chalk`{green No changes detected in any package. No test tasks is created }`
```
Comment (hsubox76) [inline scripts/ci-test/run_changed.ts:161]: "No test tasks created." (remove "is")

File: .github/workflows/test-changed-firestore.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        sudo apt-get update
+        sudo apt-get install google-chrome-stable
+    - name: Test setup and yarn install
+      run: |
+        cp config/ci.config.json config/project.json
+        yarn
+    - name: yarn build
+      run: yarn build
+    - name: Run tests if firestore or its dependenceies has changed
+      run: xvfb-run yarn test:changed:firestore
```
Comment (hsubox76) [inline .github/workflows/test-changed-firestore.yml:31]: Does firestore require xvfb-run? As long as it doesn't hurt I guess.

File: yarn.lock [hunk 8]
```diff
@@ -7,7 +7,7 @@
   resolved "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.10.4.tgz#168da1a36e90da68ae8d49c0f1b48c7c6249213a"
   integrity sha512-vG6SvB6oYEhvgisZNFRmRCUkLz11c7rp+tbNTynGqc6mS1d5ATd/sGyV6W0KZZnXRKMTzZDRgQT3Ou9jhpAfUg==
   dependencies:
-    "@babel/highlight" "^7.10.4"
+    "@babel/highlight" "^7.10.4"
```
[the remark this answers is not in the document]
Comment (Feiyang1) [reply yarn.lock:10]: Testing only. To trigger full test run. Will revert.

**Comment by Feiyang1** [comment]
@hsubox76 Created more workflows to speed up CI. Also changed the build step to only build packages needed for testing. Can you please take another look?

File: scripts/ci-test/run_changed.ts (hunk 6 above)
Comment (Feiyang1) [reply scripts/ci-test/run_changed.ts]: Done

File: .github/workflows/test-changed-firestore.yml (hunk 7 above)
Comment (Feiyang1) [reply .github/workflows/test-changed-firestore.yml]: I think they are only needed for tests that use headful browsers. Removed unnecessary xvfb-runs.

File: scripts/ci-test/build_changed.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+const config = testConfig[inputTestConfigName]!;
+
+buildForTests(config, buildAppExp);
+
+async function buildForTests(config: TestConfig, buildAppExp = false) {
+  try {
+    const testTasks = filterTasks(await getTestTasks(), config);
+    console.log(testTasks);
```
Comment (hsubox76) [inline scripts/ci-test/build_changed.ts:62]: remove? or label it?

Comment (Feiyang1) [reply scripts/ci-test/build_changed.ts]: Removed

**Pull request merged**